File: api/Altaliza.Data/Models/Veiculo.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace Altaliza.Data.Models
{
    public partial class Veiculo
    {
        public Veiculo()
        {
            PersonagemVeiculos = new HashSet<PersonagemVeiculo>();
        }

        public int Id { get; set; }
        public string Nome { get; set; }
        public int CategoriaId { get; set; }
        public int Estoque { get; set; }
        public string Imagem { get; set; }
        public decimal Preco1Dia { get; set; }
        public decimal Preco7Dia { get; set; }
        public decimal Preco15Dia { get; set; }

        public virtual Categoria Categoria { get; set; }
        public virtual ICollection<PersonagemVeiculo> PersonagemVeiculos { get; set; }
    }
}

File: web/src/models/Veiculo.ts
import { Categoria } from "./Categoria";

export interface Veiculo {
  id: number;
  nome: string;
  categoriaId: number;
  estoque: number;
  imagem: string;
  preco1dia: number;
  preco7dia: number;
  preco15dia: number;
  categoria: Categoria;
}
